packages/webdriverio/src/utils/findStrategy.ts

```diff
@@ -218,6 +222,11 @@ export const findStrategy = function (selector: SelectorStrategy, isW3C?: boolea
         value = fs.readFileSync(stringSelector, { encoding: 'base64' })
         break
     }
+    case 'role': {
+        using = 'css selector'
+        value = createRoleBaseXpathSelector(stringSelector.replace(/^\[role=/, '').replace(/]$/, ''))
```

Comment:
Hey @christian-bromann  Any specific reason to use Match because while defining strategy itself we have used match to identify the pattern and this part is used just to extract specific string(ie: role which user provided) from the selectorString

Reply:
done

Make sense to use match and return error incase of no match